packages/shared_preferences/shared_preferences_windows/lib/shared_preferences_windows.dart

```diff
@@ -100,26 +101,31 @@ class SharedPreferencesWindows extends SharedPreferencesStorePlatform {
 
   @override
   Future<bool> clear() async {
-    return clearWithPrefix(_defaultPrefix);
+    return clearWithOptions(PreferenceOptions(prefix: _defaultPrefix));
   }
 
   @override
-  Future<bool> clearWithPrefix(String prefix) async {
+  Future<bool> clearWithOptions(PreferenceOptions options) async {
     final Map<String, Object> preferences = await _readPreferences();
-    preferences.removeWhere((String key, _) => key.startsWith(prefix));
+    preferences.removeWhere((String key, _) =>
+        key.startsWith(options.prefix) &&
+        (options.allowList == null || options.allowList!.contains(key)));
     return _writePreferences(preferences);
   }
 
   @override
   Future<Map<String, Object>> getAll() async {
-    return getAllWithPrefix(_defaultPrefix);
+    return getAllWithOptions(PreferenceOptions(prefix: _defaultPrefix));
   }
 
   @override
-  Future<Map<String, Object>> getAllWithPrefix(String prefix) async {
+  Future<Map<String, Object>> getAllWithOptions(
+      PreferenceOptions options) async {
     final Map<String, Object> withPrefix =
         Map<String, Object>.from(await _readPreferences());
-    withPrefix.removeWhere((String key, _) => !key.startsWith(prefix));
+    withPrefix.removeWhere((String key, _) =>
+        !key.startsWith(options.prefix) ||
+        (options.allowList != null && !options.allowList!.contains(key)));
```

Comment:
Optional: This could be written slightly more safely (since it doesn't require a force-unwrap) as `!(key.startsWith(options.prefix) && (options.allowList?.contains(key) ?? true))`.